### DISCUSSION TITLE
请问下，我如何控制 Entity实体让他在某些情况下show属性切换？

### DISCUSSION THREAD

**Discussion by KKindom** [zouyaoji/vue-cesium#237] [Q&A] [answered]
我在地图上设置了2个实体，一个里面是很多线段，一个里面很多点。我现在想控制点击时显示或者隐藏线段。
我在实体内使用ref 获得对象 ，但是无法修改show值，每次修改报错
vue3 'set' on proxy: trap returned falsish for property show
请问下，是不是我操作的方式错误了，我给Entity设置:show="kk",然后修改kk从true到false，也没有效果。
请问下如何做到点击显示或者隐藏部分实体？
谢谢！！！！！

**Comment by KKindom** [comment, marked as answer]
我发现我的问题了，我重新设置线段id，通过 viewer.entities.getById("rain").show = false; 循环设置show属性即可。
但是我想问下大家，实体数量很多的情况下，反应会慢很多，就是过几秒才会有反应。
我时1050ti，这种大量的实体需要隐藏的话，怎么处理性能会好一些？

**Reply by zouyaoji** [reply]
放到 DataSource，然后给DataSource设置show看看好些不哇。

**Reply by KKindom** [reply]
好的我去试一试。
